.travis.yml

```diff
@@ -150,3 +150,9 @@ install:
 
 script:
 - tox -v
+# test that install and uninstall from source works
```

Comment:
This shouldn't be mangled into all jobs but should be its own job wrapped into a separate toxenv.

Reply:
We need to test this with all supported python versions but I also want to avoid exploding the number of environments we have (extra footprint). For the moment this should do the trick, we can improve it later. Lets not forget we are trying to fix an almost critical bug #590, not rewrite the entire product. I will be more than happy to add extra improvement latest if you can review them.

Reply:
FWIW the quality of this PR is unacceptable, it breaks compatibility with supported envs and mixes up several completely separate changes. So this at least should go into existing testenv in tox where it belongs, not into a selective CI service config.